src/CommandingOfficers/Commander.java

```diff
@@ -191,15 +202,63 @@ public UnitModel getUnitModel(UnitEnum unitType)
     return (unitProductionByTerrain.get(buyLocation) != null) ? unitProductionByTerrain.get(buyLocation) : new ArrayList<UnitModel>();
   }
 
-  public ArrayList<String> getReadyAbilities()
+  /** Return an ArrayList containing every ability this Commander currently has the power to perform. */
+  public ArrayList<CommanderAbility> getReadyAbilities()
   {
-    System.out.println("WARNING! Calling getReadyAbilities on Commander base class!");
-    ArrayList<String> coas = new ArrayList<String>();
-    return coas;
+    ArrayList<CommanderAbility> ready = new ArrayList<CommanderAbility>();
+    if( myActiveAbilityName.isEmpty() )
+    {
+      for( CommanderAbility ca : myAbilities )
+      {
+        if( ca.getCost() <= myAbilityPower )
+        {
+          ready.add(ca);
+        }
+      }
+    }
+    return ready;
   }
 
-  public void doAbility(String abilityName, GameInstance game)
+  public int[] getAbilityCosts()
   {
-    System.out.println("WARNING! Calling doAbility on Commander base class!");
+    int[] costs = new int[myAbilities.size()];
+    for( int i = 0; i < myAbilities.size(); ++i )
+    {
+      costs[i] = myAbilities.get(i).getCost();
+    }
+    return costs;
+  }
+
+  public double getAbilityPower()
+  {
+    return myAbilityPower;
+  }
+
+  public String getActiveAbilityName()
+  {
+    return myActiveAbilityName;
+  }
+
+  /** Lets the commander know that he's using an ability,
+   *  and accounts for the cost of using it. */
+  public void activateAbility(CommanderAbility ability)
+  {
+    modifyAbilityPower(-ability.getCost());
+    myActiveAbilityName = ability.myName;
+  }
+
+  protected void modifyAbilityPower(double amount)
```

Comment:
I feel like maybe the amount of charge the commander can hold in total should be an independent stat, rather than "whatever the most expensive one is"
This should work as well, though.

Reply:
Yeah, I considered that, but it seems a bit silly to be able to stockpile more power than you can use. I could certainly contrive a use case if I really wanted to... but eh. This is sane and simple.